pkg/v3/plugin/ocr3.go

```diff
@@ -70,8 +78,40 @@ func (plugin *ocr3Plugin[RI]) Outcome(outctx ocr3types.OutcomeContext, query typ
 	panic("ocr3 Outcome not implemented")
 }
 
-func (plugin *ocr3Plugin[RI]) Reports(seqNr uint64, outcome ocr3types.Outcome) ([]ocr3types.ReportWithInfo[RI], error) {
-	panic("ocr3 Reports not implemented")
+func (plugin *ocr3Plugin[RI]) Reports(seqNr uint64, raw ocr3types.Outcome) ([]ocr3types.ReportWithInfo[RI], error) {
+	var (
+		reports []ocr3types.ReportWithInfo[RI]
+		outcome ocr2keepersv3.AutomationOutcome
+		err     error
+	)
+
+	if outcome, err = ocr2keepersv3.DecodeAutomationOutcome(raw); err != nil {
+		return nil, err
+	}
+
+	toPerform := []ocr2keepers.CheckResult{}
+	var gasUsed uint64
+
+	for i, result := range outcome.Performable {
+		if gasUsed+result.GasUsed > plugin.ReportGasLimit || len(toPerform) > plugin.MaxUpkeepBatchSize {
```

Comment:
Just to confirm, is it possible that in one call of this Reports function, gasUsed and len(toPerform) never satisfy the above two checks, so it will never come to this if hence reports not populated properly.

Reply:
Good catch